### ISSUE TITLE
Add an example that uses directed graphs in Mesa

### ISSUE THREAD

**Issue by jackiekazil** [projectmesa/mesa#667]
RE: #653 

Seems like we need to add an example of directed graphs to Mesa. This might be adjusting how the network module works as well. 

You could do this one of two ways. 

1. You could create a directed version of the existing virus on a network, which would mean that the virus could only move in certain directions. 

2. You could recreate this model: http://ccl.northwestern.edu/netlogo/models/DiffusiononaDirectedNetwork 

It seems like #1 would be an easier lift, but I will let you decided what you prefer.

**Comment by kennethells** [comment]
I'll work on this.

**Comment by rht** [comment]
Diffusion on a directed network's code: https://github.com/NetLogo/models/blob/master/Sample%20Models/Networks/Diffusion%20on%20a%20Directed%20Network.nlogo
(it's kind of hard to parse because everything is in 1 file without boundaries)

I think it would cover more systems by porting 2) instead.

**Comment by rozenlicht** [comment]
I've chipped in by creating a PR that adds arrowheads to the D3.js visualization when `"directed": True` is added to edge in the `portrayal` dict inserted in `NetworkModule`. If this is the desired behavior, I could also code it for sigma. 
As the initial issue suggested implementing a "directed virus network" example, I've included this as well.

**Comment by rht** [comment]
This should be very easy to do with the new pure Python/Solara frontend, since it is essentially doing a NetworkX draw.

**Comment by rht** [comment]
@ankitk50 this is another low hanging fruit for warming up. The hardest part would be to find a model that behaves differently with a directed network instead of undirected network. This doesn't have to be from that NetLogo model I linked above.

**Comment by ankitk50** [comment]
this sounds interesting, but also complex, I can add this to my list.

**Comment by rht** [comment]
I think one of the version proposed by @jackiekazil should do:

> You could create a directed version of the existing [virus on a network example](https://github.com/projectmesa/mesa/tree/master/examples/virus_on_network), which would mean that the virus could only move in certain directions.

So it is not going to be a drastic modification over virus on a network.

**Comment by tpike3** [comment]
Closing to make issues more impactful and relevant to current Mesa issues (anything before Mesa 3.0 release Nov 9, 2024) per May 26, 2026 dev meeting.
